app/domain/authentication/authn_k8s/application_identity.rb

```diff
@@ -45,11 +45,25 @@ def constraints
       end
 
       def container_name
-        annotation_name = "authentication-container-name"
-        annotation_value("authn-k8s/#{@service_id}/#{annotation_name}") ||
-          annotation_value("authn-k8s/#{annotation_name}") ||
-          annotation_value("kubernetes/#{annotation_name}") ||
-          "authenticator"
+        @container_name ||= annotation_value("authn-k8s/#{@service_id}/#{authentication_container_name}") ||
+          annotation_value("authn-k8s/#{authentication_container_name}") ||
+          annotation_value("kubernetes/#{authentication_container_name}") ||
+          default_authentication_container_name
+      end
+
+      def authentication_container_name
+        "authentication-container-name"
+      end
+
+      def default_authentication_container_name
+        "authenticator".tap do |default_container_name|
```

Comment:
I don't think the tap is necessary here.

I would consider making a class constant for `default_container_name` as well (e.g. `DEFAULT_AUTHN_CONTAINER_NAME='authenticator'`). Then changing this to:

```ruby
def default_authentication_container_name
  Rails.logger.debug(
    LogMessages::Authentication::ContainerNameAnnotationDefaultValue.new(
      authentication_container_name,
      DEFAULT_AUTHN_CONTAINER_NAME
    )
  )

  DEFAULT_AUTHN_CONTAINER_NAME
end
```

Reply:
you are correct :)